lib/vector-core/src/event/value.rs

```diff
@@ -22,6 +23,41 @@ pub enum Value {
     Null,
 }
 
+impl Eq for Value {}
+
+impl Hash for Value {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match self {
+            Value::Array(v) => {
+                v.hash(state);
+            }
+            Value::Boolean(v) => {
+                v.hash(state);
+            }
+            Value::Bytes(v) => {
+                v.hash(state);
+            }
+            Value::Float(v) => {
+                if v.is_finite() {
+                    format!("{}", v).hash(state);
```

Comment:
I think it's reasonable to expect that `Hash` for `Value` will be called in hot paths. Ignoring concerns about the hashability of floats I am opposed to having `format!` here, considering it implies allocation.